charts/kyuubi/templates/kyuubi-configmap-hadoop.yaml

```diff
@@ -0,0 +1,32 @@
+{{/*
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/}}
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Release.Name }}-hadoop
+  labels:
+    {{- include "kyuubi.labels" . | nindent 4 }}
+data:
+  {{- with .Values.hadoopConf.coreSite }}
+  core-site.xml: |
+    {{- tpl . $ | nindent 4 }}
+  {{- end }}
+  {{- with .Values.hadoopConf.hdfsSite }}
+  hdfs-site.xml: |
```

Comment:
Actually, it's likely to put additional configuration files under HADOOP_CONF_DIR in practice, for example, `hive-site.xml`, `hbase-site.xml`, so we may want to allow the user to inject custom files.

Another concern is that embedded large-XML into YAML may be not user-friendly, it's easy to mess up the indent.

Reply:
Definitely agree. The approach I often see is providing both options: either provide a (templated) inline config - which as you said can get unwieldy fast - or provide the name of an existing config map. The latter allows creating that config map from a file or some other way, without relying on functionality in the chart.

Reply:
@chgl are there examples of some projects that have similar cases?

Reply:
The ones that I am aware of: 

- https://github.com/bitnami/charts/blob/main/bitnami/zookeeper/values.yaml#L203-L214, - https://github.com/bitnami/charts/blob/main/bitnami/postgresql/values.yaml#L293-L309. 
- Superset: https://github.com/apache/superset/blob/master/helm/superset/values.yaml#L135 (fairly extreme since entire python snippets are inlined) with https://github.com/apache/superset/blob/master/helm/superset/values.yaml#L50 as an option to reference an existing secret.
- akhq: https://github.com/tchiotludo/akhq/blob/dev/helm/akhq/values.yaml#L36-L46